vernac/g_vernac.mlg

```diff
@@ -440,15 +441,15 @@ GRAMMAR EXTEND Gram
       | id = identref; ":="; kind = scheme_kind -> { (Some id,kind) } ] ]
   ;
   scheme_kind:
-    [ [ IDENT "Induction"; "for"; ind = smart_global;
-          IDENT "Sort"; s = sort_family-> { InductionScheme(true,ind,s) }
-      | IDENT "Minimality"; "for"; ind = smart_global;
-          IDENT "Sort"; s = sort_family-> { InductionScheme(false,ind,s) }
-      | IDENT "Elimination"; "for"; ind = smart_global;
-          IDENT "Sort"; s = sort_family-> { CaseScheme(true,ind,s) }
-      | IDENT "Case"; "for"; ind = smart_global;
-          IDENT "Sort"; s = sort_family-> { CaseScheme(false,ind,s) }
-      | IDENT "Equality"; "for" ; ind = smart_global -> { EqualityScheme(ind) } ] ]
+    [ [ IDENT "Induction";   "for"; sch_qualid = smart_global; IDENT "Sort"; sch_sort = sort_family
+          -> { {sch_type = SchemeInduction;   sch_qualid; sch_sort} }
+      | IDENT "Minimality";  "for"; sch_qualid = smart_global; IDENT "Sort"; sch_sort = sort_family
+          -> { {sch_type = SchemeMinimality;  sch_qualid; sch_sort} }
+      | IDENT "Elimination"; "for"; sch_qualid = smart_global; IDENT "Sort"; sch_sort = sort_family
+          -> { {sch_type = SchemeElimination; sch_qualid; sch_sort} }
+      | IDENT "Case";        "for"; sch_qualid = smart_global; IDENT "Sort"; sch_sort = sort_family
+          -> { {sch_type = SchemeCase;        sch_qualid; sch_sort} }
+    ] ]
```

Comment:
How about refactoring: create a new nonterminal that captures the alternatives Induction, Minimality, etc. and then share the common part:

```
sch_type = some_new_nonterminal; "for"; sch_qualid = smart_global; IDENT "Sort"; sch_sort = sort_family
          -> { {sch_type; sch_qualid; sch_sort} }
```

Reply:
Alright done